lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2490,24 +2490,66 @@ def delayed_completion(self) -> Delayed:
         result can be computed, otherwise computation will hang (never return).
         """
         if self._delayed_writes:
-            # Create a single delayed da.store operation to complete the file.
+            # Create a single delayed operation to complete the file.
             sources, targets, fill_infos = zip(*self._delayed_writes)
-            store_op = da.store(sources, targets, compute=False, lock=False)
 
-            # Construct a delayed fill-check operation for each (lazy) source array.
-            delayed_fillvalue_checks = [
-                # NB with arraylib=dask.array, this routine does lazy array computation
-                _data_fillvalue_check(da, source, fillinfo.check_value)
-                for source, fillinfo in zip(sources, fill_infos)
-            ]
+            # Define a combined store-and-fillvalue-check operation, delayed for
+            #  application to lazy data blocks.
+            @dask.delayed
+            def delayed_block_save_and_fv_check(
+                datablock, target_block, block_slices, fillinfo
+            ):
+                # The inner function of a delayed fill-value check
+                # Save the block and return array([is_masked, contains_value]).
+                target_block[block_slices] = datablock
+                is_masked, contains_value = _data_fillvalue_check(
+                    np, datablock, fillinfo.check_value
+                )
+                return np.array([is_masked, contains_value])
+
+            # Map the function across the blocks of the input data.
+            # NOTE: the goal is to process the to-be-saved blockwise via a delayed
+            # operation that will return (is_masked, contains_value) for each block.
+            # These block-wise results are then OR-ed for each source.
+            # N.B. `da.map_blocks` apparently will not do this, at least not without
+            #  `drop_axis` which seems to trigger loading of the whole source array.
+            # TODO: It probably *can* be done with `da.blockwise`, but I have so far
+            #  failed to grasp how you would do that.
```

Comment:
This looks like a [reduction](https://docs.dask.org/en/latest/generated/dask.array.reduction.html), maybe that could be useful?

Reply:
:heart: :think: